mautic/mautic
# Toogle for public preview of email

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | 
| Automated tests included? |
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | https://github.com/mautic/mautic/issues/6668
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This PR added option to enable/disable public view of email preview. 
New decent toggle option added before public link.

![image](https://user-images.githubusercontent.com/462477/46506064-61882600-c833-11e8-9c80-456072f00b24.png)

Also this update allow developers  use this toggle button anywhere for any boolean type:

- example: https://github.com/mautic/mautic/pull/6679/files#diff-3533e31fe48dc7cc3fcadb14a34007b1R311
- just use customToggle in param `'query' => 'customToggle=publicPreview'`
- your entity require getter and setter for that parameter

#### Steps to test this PR:
1.  Create published email
2.  Copy preview link and open it in another/incognito brower
3.  Should see  preview of email
4. Go to detail page of email and disable toggle before preview link 
![image](https://user-images.githubusercontent.com/462477/46506240-db201400-c833-11e8-88be-2a661886d031.png)
5.  Copy preview link and open it in another/incognito brower
6. Should see access denied page
